doc/design/test-dandisets-1.md

```diff
@@ -0,0 +1,102 @@
+assets. These dandisets are a subset of draft-only dandisets, since publishing a
+dandiset requires assets to be present.
+
+This option is problematic on its own--it is a valid use case to declare
+scientific intentions by creating a draft dandiset with no assets, then collect
+and add data only after months or years of real scientific work. That is to say,
+an assetless dandiset that nonetheless has a description, funding agent,
+contact person, owners, etc. may indeed be fully valid.
+
+### Add questions to dandiset creation page
+
+Adding questions such as:
+- Do you welcome collaborators? (Y/N)
+- When do you anticipate posting data to the dandiset?
+
+may help to stem the tide of abandoned dandisets.
+
+The collaborators question, answered positively, would add a badge to the
+dandiset's landing page and search result display indicating the call for
+collaboration.
+
+The anticipation date question would delay considering the dandiset abandoned
+until after that date.
+
+### Enable deletion of dandisets
+
+Part of the problem may be that people simply cannot delete their own abanonded
+dandisets. Offering a delete button with the proper semantics could help here.
+
+The archive does not allow for deletion of *published* dandisets, by design. But
+dandisets that are draft-only should be deleteable.
+
+And, as en extension to that idea, dandisets that are published should be able
+to have their draft dandiset deleted (essentially "archiving" that dandiset, and
+closing it off to further updates).
+
+### Approval and cleanup process for dandisets
+
+On a regular basis, collect dandisets that are assetless, draft-only, or
+otherwise seemingly invalid, and email the owners thereof. Inform them that the
+dandiset will be deleted in some reasonable amount of time unless they take
+specific actions, which could be something like:
+- delete the dandiset
+- add assets to the dandiset
+- respond to the email to explain why the dandiset is actually valid
+
+We would need to fine tune the criteria for "invalid". For example, adding
+assets to a dandiset is great, but if thereafter the dandiset does not get
+published in some timeframe (e.g., six months), then perhaps it is considered
+invalid again.
+
+The final action listed above would put the dandiset through a sort of
+pre-emptive approval process. If the owner explains that indeed they have
+collaborators, they are collecting data, etc., but it will take a while, the
+admins could put a timer on the dandiset for when to check in again, etc.
+
+This is just a sketch of how this feature might work. It is extremely complex,
+requiring policy, careful communication, and buy-in from the community. It would
+also take focused labor from the admin team, and there are risks associated with
+non-communicative dandiset owners who may become frustrated with having their
+dandisets deleted, etc. But on the plus side, it would also bring richness and
+value to the dandisets hosted on the archive.
```

Comment:
Maybe instead of deleting invalid dandisets, they can simply be "archived". This would remove the dandiset from all public facing pages, and it would only be viewable by the owners in their "My Dandisets" page. Upon a dandiset being archived, all owners of the dandiset would receive an email alerting them of it. The only thing an owner needs to do to unarchive it is to login to dandi and click "unarchive" on the dandiset.

This also has the benefit of being completely automatable.

Reply:
I agree though I think the language might be confusing, just because DANDI calls itself an "archive." By that definition, wouldn't all dandisets be "archived?" It might not be clear what "archiving" would do (though I understand it would be similar to how gmail uses the term)

Reply:
Ah good point, yes I agree "delist" would be better than "archive".

> On the other hand, this might be abused to use DANDI resources to create private dandisets on the sly..

If a dandiset has assets it wouldn't be considered an "invalid" dandiset and thus couldn't be delisted. And a "private" dandiset without any assets doesn't have much use to people looking to abuse the system. 

We could also make delisted dandisets readonly until they are relisted by an owner.

Reply:
oh, ok. Sure, we can only allow assetless dandisets to be delisted. Or we could allow any dandiset to be delisted, and allow users to view delisted dandisets in advanced settings.

Reply:
I love the delisting idea, including making delisted dandisets read-only (to prevent abuse).

I also think there could be a reason to delist dandisets with assets in them--specifically, draft-only dandisets with, e.g., incomplete assets aren't useful to the community, so at some point it's gotta be "publish or perish" so to speak.

Reply:
delisting sounds like the `"Test" flag` section, but with a different label. I think it would make sense for that to be a boolean setting available for all draft dandisets that can be set during creation and updated later on by owners or admins.

Reply:
I toyed [above](https://github.com/dandi/dandi-api/pull/606#discussion_r779652437) with the idea of making the "test" flag autodelete the dandiset after 7 days, so that if a user just wants to try something out they can sort of fire and forget. Then, the "delisting" concept would apply more broadly to non-test dandisets as discussed in this thread.

Reply:
I significantly reworked this section to focus on delisting instead; see 5b8ed7112466876c9992e4d41f6720d6500cb9ce. There's a lot to discuss here, so please give me feedback.

(In particular, the time ranges I suggested for delisting criteria are arbitrary and probably too short.)